sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/hints.scala

```diff
@@ -44,11 +45,27 @@ case class ResolvedHint(child: LogicalPlan, hints: HintInfo = HintInfo())
   override def doCanonicalize(): LogicalPlan = child.canonicalized
 }
 
+/**
+ * Hint that is associated with a [[Join]] node, with [[HintInfo]] on its left child and on its
+ * right child respectively.
+ */
+case class JoinHint(
```

Comment:
The indentation is wrong here

Reply:
https://github.com/databricks/scala-style-guide#spacing-and-indentation